src/bson_mapper/CMakeLists.txt

```diff
@@ -119,6 +119,13 @@ install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
         REGEX ".+/test" EXCLUDE
 )
 
+install(DIRECTORY ${CMAKE_SOURCE_DIR}/src/third_party/cereal/include/cereal/
+    DESTINATION "include/cereal"
```

Comment:
I think you should change this to 
`DESTINATION ${BSON_MAPPER_HEADER_INSTALL_DIR}/bson_mapper/third_party/cereal`

and change Cflags in libbson_mapper.pc.in to:
`Cflags: -I${includedir}/bson_mapper/@BSON_MAPPER_INLINE_NAMESPACE@ -I${includedir}/bson_mapper/@BSON_MAPPER_INLINE_NAMESPACE@/bson_mapper/third_party`

This way the cereal headers will be included in the pkg-config for libbson_mapper